code/game/machinery/computer/shuttle_computers.dm

```diff
@@ -362,6 +385,39 @@
 			to_chat(usr, "Successfully linked [src] to [capitalize(S.name)]!")
 			src.updateUsrDialog()
 
+	if(href_list["custom_coord"])
+		switch(href_list["custom_coord"])
+			if("x")
+				custom_x = input("Enter new X drift", "Course Plotting", custom_x) as num
+			if("y")
+				custom_y = input("Enter new Y drift", "Course Plotting", custom_y) as num
+			if("z")
+				custom_z = input("Enter new Z drift", "Course Plotting", custom_z) as num
+			if("a")
+				custom_rot=input("Enter rotation angle", "Course Plotting", custom_rot) as num
+
+		src.updateUsrDialog()
+
+	if(href_list["process_custom_coord"])
+		if(istype(disk, /obj/item/weapon/disk/shuttle_coords/free_move))
+			var/turf/dest = locate(\
+			shuttle.linked_port.x + custom_x,\
+			shuttle.linked_port.y + custom_y,\
+			shuttle.linked_port.z + custom_z
+			)
+
+			if(!dest)
+				to_chat(usr, "Error! Bad coordinates.")
+				return
+
+			disk.destination = new /obj/docking_port/destination(dest)
```

Comment:
Is this ever deleted or does it keep creating a new docking_port each time it travels?